sycl/include/CL/sycl/detail/kernel_program_cache.hpp

```diff
@@ -25,6 +25,11 @@ namespace detail {
 class context_impl;
 class KernelProgramCache {
 public:
+  struct BuildResultT {
+    std::string Msg;
+    cl_int Code;
```

Comment:
You probably need to use `pi_result` here.

Reply:
pi_result is an enum. 
Exception return cl_int, which is a mere int.
Assigning enum from int is prohibited.

Reply:
pi_int32 then?

Reply:
OK.